core_lib/src/util/pencilerror.h

```diff
@@ -90,8 +90,6 @@ class Status
     bool operator==(ErrorCode code) const;
     bool operator!=(ErrorCode code) const;
 
-    void updateStatus(const ErrorCode code) { mCode = code; }
```

Comment:
Hmm I think this is decent convenience to have instead of constructing a new status object every time you need to update it.. I can't justify the change in this PR anymore though given the changes you made.

Reply:
Tbh, normally I would totally agree. But in this case we’re essentially talking about a *return* status. So if the code is well-structured, you shouldn’t really have a reason to update the status code IMO. But as I said, that’s kinda specific to this particular use case.